### DISCUSSION TITLE
Is there a way to disable write cache on NVMe devices using udisks?

### DISCUSSION THREAD

**Discussion by TomaszGasior** [storaged-project/udisks#1240] [Q&A] [answered]
NVMe support has been added recently to Udisks daemon but there are no NVMe-specific settings in `man udisks`. I would like to disable write cache on my disk using Udisks. Is it possible? Or will it be possible in the future?

**Comment by vojtechtrefny** [comment]
We currently don't support that. The NVMe support is for now limited to SMART (health and self tests) and secure erase. I am not sure about the future plans, @tbzatek?

**Comment by tbzatek** [comment, marked as answer]
NVMe support in UDisks is currently limited to a couple of features and the intention has always been to add new features per "as-needed" basis. We just need a feedback/request just like you do here. I'll put this on a TODO list then, the API might be perhaps similar to the `org.freedesktop.UDisks2.Drive.Ata` interface.

However, what we're not going to do, is to provide extensive NVMe functionality in `udisksctl`. You should be using `nvme-cli` directly for that, it's the common code (provided by `libnvme`) behind that after all.

**Reply by TomaszGasior** [reply]
There are two reasons why I am asking the question. First, write cache feature is not persisted, at least in my disk's controller. This means I have to `sudo nvme set-feature /dev/nvme0 -f 6 -v 0` after each boot up and also after each resume from suspend. Having this ugly work already done by udisks would very nice. Second, it would be nice to have write cache setting available in GNOME Disks, the same way as currently for SATA disks. I am 100% sure Disks maintainers will not be interested in creating and maintaining the implementation of invoking `sudo nvme set-feature /dev/nvme0 -f 6 -v 0` on each boot up and each resume. Udisks would be much better place for that.

For now I don't see a reason for having other NVMe options available for Udisks.

**Reply by TomaszGasior** [reply]
For context: being able to disable write cache seems to be a workaround for faulty disks combined with BTRFS, which is the default file system for Fedora.  That's how I went here.

https://wiki.tnonline.net/w/Btrfs/Parent_Transid_Verify_Failed
